### DISCUSSION TITLE
Gzip support (permessage-deflate)

### DISCUSSION THREAD

**Discussion by intellix** [enisdenjo/graphql-ws#516] [Q&A] [answered]
I'm not sure if this is the right place to ask about support, but I'm wondering if it's possible to do gzipping in the websocket layer and if graphql-ws supports it

**Comment by enisdenjo** [comment, marked as answer]
[WebSockets do support compression.](https://datatracker.ietf.org/doc/html/rfc7692) However, this happens one layer above `graphql-ws` so we inherently support compression if the server does.

If you're using Node `ws` library, you need to set up the server configuring [the `perMessageDeflate` options](https://github.com/websockets/ws#websocket-compression).

**Reply by intellix** [reply]
awesome, does this just work in the client as well?

**Reply by enisdenjo** [reply]
The browser should handle that for you automatically. No configuration needed there either.
